src/Microsoft.Extensions.Logging/ConfigurationLoggerFilterConfigureOptions.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.Logging
+{
+    public class ConfigurationLoggerFilterConfigureOptions : IConfigureOptions<LoggerFilterOptions>
+    {
+        private readonly IConfiguration _configuration;
+
+        public ConfigurationLoggerFilterConfigureOptions(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        public void Configure(LoggerFilterOptions options)
+        {
+            LoadDefaultConfigValues(options.Rules);
+        }
+
+        private void LoadDefaultConfigValues(ICollection<LoggerFilterRule> rules)
+        {
+            if (_configuration == null)
+            {
+                return;
+            }
+
+            foreach (var configurationSection in _configuration.GetChildren())
+            {
+                if (configurationSection.Key == "LogLevel")
+                {
+                    // Load global category defaults
+                    LoadRules(rules, configurationSection, null);
+                }
+                else
+                {
+                    var logLevelSection = configurationSection.GetSection("LogLevel");
+                    if (logLevelSection != null)
+                    {
+                        // Load logger specific rules
+                        var logger = ExpandLoggerAlias(configurationSection.Key);
+                        LoadRules(rules, logLevelSection, logger);
+                    }
+                }
+            }
+        }
+
+        private static void LoadRules(ICollection<LoggerFilterRule> rules, IConfigurationSection configurationSection, string logger)
+        {
+            foreach (var section in configurationSection.AsEnumerable(true))
+            {
+                if (TryGetSwitch(section.Value, out var level))
+                {
+                    rules.Add(new LoggerFilterRule(logger, section.Key, level, null));
+                }
+            }
+        }
+
+        public string ExpandLoggerAlias(string name)
```

Comment:
This doesn't seem like a scalable API; it'll also deter users from stepping beyond these basic providers because the natural guess will be wrong. I.e., if I'm a user looking at an existing configuration, and want to add a new one for, say, a file, I'll try `"File"` and nothing will happen... It also seems a bit questionable to favor, say, Azure App Services, but not Seq, Splunk, Elasticsearch, CloudWatch....